app/queries/student_details_resolver.rb

```diff
@@ -0,0 +1,104 @@
+class StudentDetailsResolver < ApplicationQuery
+  property :student_id
+
+  def student_details
+    {
+      name: student.name,
+      title: student.title,
+      email: student.email,
+      avatar_url: avatar_url,
+      phone: student.phone,
+      coach_notes: coach_notes,
+      targets_completed: targets_completed,
+      total_targets: total_targets,
+      level_id: level.id,
+      social_links: social_links,
+      evaluation_criteria: evaluation_criteria,
+      quiz_scores: quiz_scores,
+      average_grades: average_grades,
+      levels_completed: levels_completed
+    }
+  end
+
+  def levels_completed
+    required_targets_by_level = Target.joins(:target_group).where(target_groups: { milestone: true, level_id: course.levels.select(:id) }).distinct(:id)
+      .pluck(:id, 'target_groups.level_id').each_with_object({}) do |(target_id, level_id), required_targets_by_level|
+      required_targets_by_level[level_id] ||= []
+      required_targets_by_level[level_id] << target_id
+    end
+
+    passed_target_ids = TimelineEvent.joins(:founders).where(founders: { id: student.id }).where.not(passed_at: nil).distinct(:target_id).pluck(:target_id)
+
+    course.levels.pluck(:id).select do |level_id|
+      ((required_targets_by_level[level_id] || []) - passed_target_ids).empty?
+    end.map(&:to_s)
+  end
+
+  def average_grades
+    @average_grades ||= TimelineEventGrade.where(timeline_event: submissions).group(:evaluation_criterion_id).average(:grade).map do |ec_id, average_grade|
+      { id: ec_id, average_grade: average_grade.round(1) }
+    end
+  end
+
+  def targets_completed
+    submissions.passed.distinct(:target_id).count
```

Comment:
replace with  `submissions.passed.pluck(:target_id).uniq.count`

Reply:
Fixed in 49ddfe3e